fix(diff): fix prerelease to stable version diff logic

eminberkayd opened the pull request:
Refactored version diff logic to handle transitions from prerelease versions to stable versions correctly for major/minor/patch version bumps.

Closes #606

wraithgar commented:
Looking at what needed to be done to fix this, and seeing the code that was left, I had a thought that this wasn't complete and I was right.

```sh
~/D/n/n/b/main (fix/prerelease-diff-logic-#606|✔) $ node
Welcome to Node.js v22.13.1.
Type ".help" for more information.
> require('.').diff('1.1.1-pre', '2.1.1')
'minor'
> 
```

So, I think you've identified what's wrong here (order of operations) but we obviously still have holes.

wraithgar commented:
```patch
diff --git a/functions/diff.js b/functions/diff.js
index f011d99..58f9fde 100644
--- a/functions/diff.js
+++ b/functions/diff.js
@@ -27,8 +27,13 @@ const diff = (version1, version2) => {
       return 'major'
     }
 
-    // Otherwise it can be determined by checking the high version
+    // If the majors differ then it's major
+    if (highVersion.major !== lowVersion.major) {
+      return 'major'
+    }
 
+    // Otherwise it can be determined by checking the high version
+    //
     if (highVersion.minor) {
       // anything higher than a minor bump would result in the wrong version
       return 'minor'
```

```sh
> require('.').diff('1.1.1-pre', '2.1.1-pre')
'premajor'
> require('.').diff('1.1.1-pre', '2.1.1')
'major'
> 
```

Tests still pass with this, would need to add test cases for this new case.

eminberkayd commented:
I have incorporated additional test cases and updated the PR accordingly. Kindly review it when you have time, and let me know if any modifications are required. Thank you!

wraithgar commented:
Oh yes this is much better.
